### PR TITLE
Feat: fetch all billing plan

### PR DESCRIPTION
**Pull request by Ayobamidele**
## Description

This pull request introduces a new endpoint `GET /api/v1/organizations/billing-plans` that allows users to retrieve all billing plans associated with a specified organization. The endpoint responds with a JSON object containing the status of the request, a message, and an array of billing plans, each with its ID, name, price, and features.

### Example Response:

```json
{
  "status": 200,
  "message": "plans fetched successfully",
  "data": {
    "plans": [
      {
        "id": "basic",
        "name": "Basic Plan",
        "price": 10.00,
        "features": ["Feature 1", "Feature 2"]
      },
      {
        "id": "premium",
        "name": "Premium Plan",
        "price": 20.00,
        "features": ["Feature 1", "Feature 2", "Feature 3"]
      }
    ]
  }
}
```
Related Issue (Link to issue ticket)
[#321](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/321)

**Motivation and Context**

This change is required to provide organizations with the ability to view available billing plans. It addresses the need for enhanced visibility of billing options, allowing organizations to make informed decisions about subscription levels and features they may want to use.

**How Has This Been Tested?**

The new endpoint has been tested using Postman to ensure that the correct billing plans are fetched based on valid organization IDs. Testing included the following:

    

- [x] Sending requests with valid authorization tokens.
- [x] Verifying the response format, status code, and data integrity.
- [x] Ensuring that appropriate status codes are returned for unauthorized tokens

Screenshot 2024-07-24 125004
Screenshot 2024-07-24 125025

Types of changes

   - [ ] Bug fix (non-breaking change which fixes an issue)
   - [x] New feature (non-breaking change which adds functionality)
   - [ ] Breaking change (fix or feature that would cause existing functionality to change)

Checklist:

  -  [x] My code follows the code style of this project.
  -  [ ] My change requires a change to the documentation.
  -  [ ] I have updated the documentation accordingly.
  -  [x] I have read the CONTRIBUTING document.
  -  [x] I have added tests to cover my changes.
  -  [x] All new and existing tests passed.

### CODE REVIEW

**Review by joboy-dev — APPROVED** [review]

File: api/v1/models/profile.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 from api.v1.models.base import Base
 from api.v1.models.base_model import BaseTableModel
 
+# from api.v1.models.user import User
+
```
Comment (Goketech) [inline api/v1/models/profile.py:14]: uncomment this

File: api/v1/routes/auth.py [hunk 2]
```diff
@@ -131,47 +131,3 @@ def refresh_access_token(request: Request, response: Response, db: Session = Dep
...
-
-    # Send the token to the user's email
-    # send_mail(to=user.email, subject="Your SignIn Token", body=token)
-
-    return success_response(
-        status_code=200,
-        message="Sign-in token sent to email"
-    )
-
-@auth.post("/verify-token", status_code=status.HTTP_200_OK)
```
Comment (Goketech) [inline api/v1/routes/auth.py:154]: is your branch updated?

fetch from dev and commit

File: api/v1/schemas/plans.py [hunk 3]
```diff
@@ -14,4 +14,5 @@ class SubscriptionPlanResponse(CreateSubscriptionPlan):
     id: uuid.UUID
     
     class Config:
-        orm_mode = True
\ No newline at end of file
+        orm_mode = True
+
```
Comment (Goketech) [inline api/v1/schemas/plans.py:18]: revert changes

**Review by joboy-dev — CHANGES_REQUESTED** [review]
You removed requirements and all alembic migrations????

**Review by Goketech — CHANGES_REQUESTED** [review]
fetch from dev and make your branch updated

revert changes to requirements.tx and make requested changes

revert changes in alembic.ini

File: api/v1/models/profile.py (hunk 1 above)
Comment (Ayobamidele) [reply api/v1/models/profile.py:14]: that was not me

File: api/v1/routes/auth.py (hunk 2 above)
Comment (Ayobamidele) [reply api/v1/routes/auth.py:154]: that was someone else

Comment (Ayobamidele) [reply api/v1/routes/auth.py:154]: I'll merge from dev to update the code, the changes must have affected it

File: api/v1/models/profile.py (hunk 1 above)
Comment (Ayobamidele) [reply api/v1/models/profile.py:14]: I did not even touch the profile.py

**Review by joboy-dev — DISMISSED** [review]

File: api/v1/routes/auth.py [hunk 4]
```diff
@@ -167,11 +167,4 @@ async def verify_signin_token(token_schema: TokenRequest, db: Session = Depends(
...
             "token_type": "bearer"
         }
-    )
-    
-
-# Protected route example: test route
-@auth.get("/admin")
-def read_admin_data(current_admin: Annotated[User, Depends(user_service.get_current_super_admin)]):
-    return {"message": "Hello, admin!"}
-
```
Comment (Goketech) [inline api/v1/routes/auth.py:177]: revert this

**Review by Goketech — CHANGES_REQUESTED** [review]

**Pull request closed without merging**